pandas/core/categorical.py

```diff
@@ -356,8 +357,26 @@ def itemsize(self):
         """ return the size of a single category """
         return self.categories.itemsize
 
-    def reshape(self, new_shape, **kwargs):
-        """ compat with .reshape """
+    def reshape(self, new_shape, *args, **kwargs):
+        """
+        An ndarray-compatible method that returns
+        `self` because categorical instances cannot
+        actually be reshaped.
+        """
+        # while the 'new_shape' parameter has no effect,
+        # we should still enforce valid shape parameters
+        shape = np.asarray(new_shape)
```

Comment:
ok, ``.reshape`` is really just for compat, there is no use actually in pandas, but then if the shape is different we give back a numpy array. I would extract out this change and we can consider it in the future. I think we should remove that behavior (so for now just try to exactly replicate the arg checking)